src/backend/dx12ll/src/lib.rs

```diff
@@ -130,8 +136,26 @@ impl core::Adapter for Adapter {
             }).collect::<Vec<_>>()
         }).collect();
 
-        let heap_types = unimplemented!();
-        let memory_heaps = unimplemented!();
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/dn788678(v=vs.85).aspx
+        let heap_types = vec![
+            core::HeapType {
+                id: 0,
+                properties: memory::DEVICE_LOCAL,
+                heap_index: 1,
+            },
+            core::HeapType {
+                id: 1,
+                properties: memory::CPU_VISIBLE | memory::CPU_CACHED,
```

Comment:
Does D3D12 have non-coherent memory? Vulkan requires to explicitly flush non-coherent memory via `vkFlush/InvalidateMappedMemoryRanges`. I don't recall if there is a similar restriction on d3d12 (map/unmap?).